frontend/src/metabase/search/containers/SearchApp.jsx

```diff
@@ -25,172 +30,90 @@ import {
+    [location],
+  );
 
-  const query = {
-    q: location.query.q,
-    limit: PAGE_SIZE,
-    offset: PAGE_SIZE * page,
-  };
+  const searchFilters = useMemo(() => {
+    return getFiltersFromLocation(location);
+  }, [location]);
+
+  const [selectedSidebarType, setSelectedSidebarType] = useState(null);
+
+  useEffect(() => {
+    setSelectedSidebarType(null);
+  }, [searchText, searchFilters]);
+
+  const query = useMemo(
+    () => ({
+      q: searchText,
+      ..._.omit(searchFilters, SearchFilterKeys.Type),
+      models: selectedSidebarType ?? searchFilters[SearchFilterKeys.Type],
+      limit: PAGE_SIZE,
+      offset: PAGE_SIZE * page,
+    }),
+    [searchText, searchFilters, selectedSidebarType, page],
+  );
 
-  if (filter) {
-    query.models = filter;
-  }
+  const onChangeSelectedType = filter => {
+    setSelectedSidebarType(filter);
+    setPage(0);
+  };
 
   return (
-    <SearchRoot>
-      {location.query.q && (
+    <SearchRoot data-testid="search-app">
+      {searchText && (
         <SearchHeader>
-          <Subhead>{jt`Results for "${location.query.q}"`}</Subhead>
+          <Subhead>{jt`Results for "${searchText}"`}</Subhead>
         </SearchHeader>
       )}
-      <div>
-        <Search.ListLoader query={query} wrapped>
-          {({ list, metadata }) => {
-            if (list.length === 0) {
-              return (
-                <SearchEmptyState>
-                  <Card>
-                    <EmptyState
-                      title={t`Didn't find anything`}
-                      message={t`There weren't any results for your search.`}
-                      illustrationElement={<img src={NoResults} />}
-                    />
-                  </Card>
-                </SearchEmptyState>
-              );
-            }
-
-            const availableModels = metadata.available_models || [];
-
-            const filters = SEARCH_FILTERS.filter(f =>
-              availableModels.includes(f.filter),
-            );
-
-            return (
-              <SearchBody>
-                <SearchMain>
-                  <Fragment>
-                    <SearchResultSection items={list} />
-                    <div className="flex justify-end my2">
-                      <PaginationControls
-                        showTotal
-                        pageSize={PAGE_SIZE}
-                        page={page}
-                        itemsLength={list.length}
-                        total={metadata.total}
-                        onNextPage={handleNextPage}
-                        onPreviousPage={handlePreviousPage}
-                      />
-                    </div>
-                  </Fragment>
-                </SearchMain>
+      <Search.ListLoader query={query} wrapped>
+        {({ list, metadata }) =>
+          list && list.length > 0 ? (
+            <SearchBody>
+              <SearchMain>
+                <SearchResultSection items={list} />
+                <PaginationControls
```

Comment:
The pagination looks different from `master` and from the design.

## This PR
![Screenshot 2023-08-04 at 7 43 40 PM](https://github.com/metabase/metabase/assets/1937582/e6be1b28-962d-464c-8e67-4829c34bdf18)


## `master`
![Screenshot 2023-08-04 at 7 43 42 PM](https://github.com/metabase/metabase/assets/1937582/8f1778b6-d264-4170-8673-11a4a99a5b5d)

Reply:
Yep, missing some margin. I'll sort it now since it's an easy fix

Reply:
@oisincoveney the first sceenshot is now updated. I uploaded the wrong one.